internal/broker/broker.go

```diff
@@ -620,6 +613,21 @@ func (b *Broker) handleIsAuthenticated(ctx context.Context, session *session, au
 		}
 	}
 
+	if !b.userNameIsAllowed(authInfo.UserInfo.Name) {
+		return AuthDenied, errorMessage{Message: "permission denied"}
+	}
+
+	// If the owner is unset and allowed, we auto-generate a config file with the first
+	// user to login as the owner.
+	if b.cfg.userConfig.shouldRegisterOwner() {
+		if err := b.cfg.registerOwner(b.cfg.ConfigFile, authInfo.UserInfo.Name); err != nil {
```

Comment:
Just a minor question (feel free to ignore if I am wrong):

Is the check `b.cfg.userConfig.shouldRefisterOwner()` a bit duplicated with the check in the `b.cfg.registerOwner`? Is it okay to remove the check logic in the `registerOwner` method?

Reply:
It is, wanted to be extra safe that the function won't be called when it shouldn't. As it seems redundant now, I will remove it.